ui/app/src/views/ViewProject.tsx

```diff
@@ -32,36 +37,80 @@ function DashboardPageInProject(props: RenderDashboardInProjectProperties) {
   const [openCreateDashboardDialogState, setOpenCreateDashboardDialogState] = useState(false);
 
   const { data } = useDashboardList(props.projectName);
-  if (data === undefined) {
-    return null;
-  }
 
   const handleDashboardCreation = function (name: string) {
     navigate(`/projects/${props.projectName}/dashboards/${name}/create`);
   };
 
   return (
-    <Paper>
-      <Box p={1}>
-        <Stack direction="row" alignItems="center" justifyContent="space-between">
-          <Stack direction="row" alignItems="center" gap={1} my={2}>
-            <ViewDashboard />
-            <Typography variant="h3">Dashboards</Typography>
-          </Stack>
-          <Button variant="contained" size="small" onClick={() => setOpenCreateDashboardDialogState(true)}>
-            Add Dashboard
-          </Button>
+    <Box>
+      <Stack direction="row" alignItems="center" justifyContent="space-between">
+        <Stack direction="row" alignItems="center" gap={1} my={2}>
+          <ViewDashboard />
+          <Typography variant="h3">Dashboards</Typography>
         </Stack>
-        <ErrorBoundary FallbackComponent={ErrorAlert}>
-          <DashboardList dashboardList={data} />
-        </ErrorBoundary>
-        <CreateDashboardDialog
-          open={openCreateDashboardDialogState}
-          onClose={() => setOpenCreateDashboardDialogState(false)}
-          onSuccess={(name: string) => handleDashboardCreation(name)}
-        />
-      </Box>
-    </Paper>
+        <Button variant="contained" size="small" onClick={() => setOpenCreateDashboardDialogState(true)}>
+          Add Dashboard
+        </Button>
+      </Stack>
+      <ErrorBoundary FallbackComponent={ErrorAlert}>
+        <DashboardList dashboardList={data || []} hideToolbar={props.hideToolbar} />
+      </ErrorBoundary>
+      <CreateDashboardDialog
+        open={openCreateDashboardDialogState}
+        onClose={() => setOpenCreateDashboardDialogState(false)}
+        onSuccess={(name: string) => handleDashboardCreation(name)}
+      />
+    </Box>
+  );
+}
+
+interface RecentVisitedDashboardsInProjectProperties {
+  projectName: string;
+}
+
+function RecentVisitedDashboardsInProject(props: RecentVisitedDashboardsInProjectProperties) {
+  const navigate = useNavigate();
+  const history = useNavHistory();
+
+  const [openCreateDashboardDialogState, setOpenCreateDashboardDialogState] = useState(false);
+
+  const { data } = useDashboardList(props.projectName);
+  const recentDashboard = useMemo(() => {
```

Comment:
I just thought I could create a hook useRecentDashboard in dashboard-client.ts, probably more clean and can be used somewhere else

Reply:
Idea implemented + added added Card component around DataGrid to have a better look 😄 Image updated on the desc